### DISCUSSION TITLE
Fixed Time-range fro grafana variables

### DISCUSSION THREAD

**Discussion by Ahmed-Dghaies** [grafana/grafana#38281] [Dashboards, Panels & Visualizations]
Hi,
I have a dashboard variable that i’m getting its values from elastic-search and the time-range of getting them is the same time-range of the dashboard itself. However when i don’t have data logged for a specific variable it will not appear in the variable result which is not the preferred outcome
so is there anyway i can fix the time range of the query to be last 7 days as an example or last month?
here is the query i use
{“find”: “terms”, “field”: “sysloghost.keyword”, “query”: “”}

**Comment by ivanahuckova** [comment]
Just to confirm I understand this correctly - this is a feature request and you would like to have option to select time range for your template variables? So instead of them using the same time as it is in dashboard, it would be always (for example) last 7 days.

**Comment by Ahmed-Dghaies** [comment]
Hi, yes exactly

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
